core/node/state_keeper/src/updates/l2_block_updates.rs

```diff
@@ -145,11 +124,17 @@ impl L2BlockUpdates {
 
         // Get transaction factory deps
         let factory_deps = &tx.execute.factory_deps;
-        let tx_factory_deps: HashMap<_, _> = factory_deps
+        let mut tx_factory_deps: HashMap<_, _> = factory_deps
             .iter()
-            .map(|bytecode| (hash_bytecode(bytecode), bytecode))
+            .map(|bytecode| (hash_bytecode(bytecode), bytecode.clone()))
             .collect();
 
+        new_known_factory_deps
+            .into_iter()
+            .for_each(|(hash, bytecode)| {
+                tx_factory_deps.insert(hash, bytecode);
+            });
```

Comment:
Nit: This is not JavaScript; please use a `for` cycle, or `tx_factory_deps.extend(_)`.